workflow/controller/operator.go

```diff
@@ -2968,6 +2980,46 @@ func (woc *wfOperationCtx) runOnExitNode(ctx context.Context, templateRef, paren
 	return false, nil, nil
 }
 
+func (woc *wfOperationCtx) resolveExitTmplArgument(args wfv1.Arguments, prefix string, outputs *wfv1.Outputs) (wfv1.Arguments, error) {
+	scope := createScope(nil)
+	for _, param := range outputs.Parameters {
+		scope.addParamToScope(fmt.Sprintf("%s.outputs.parameters.%s", prefix, param.Name), param.Value.String())
+	}
+	for _, arts := range outputs.Artifacts {
+		scope.addArtifactToScope(fmt.Sprintf("%s.outputs.artifacts.%s", prefix, arts.Name), arts)
+	}
+
+	stepBytes, err := json.Marshal(args)
+	if err != nil {
+		return args, errors.InternalWrapError(err)
```

Comment:
I don’t like to wrap errors using this way anymore. Adds no value. Instead use fmt.Errrof.